src/main/java/org/killbill/billing/plugin/catalog/test/models/api/PlanModel.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2022-2022 The Billing Project, LLC
+ *
+ * The Billing Project licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.killbill.billing.plugin.catalog.test.models.api;
+
+import java.util.Arrays;
+import java.util.Objects;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import org.killbill.billing.catalog.api.Plan;
+import org.killbill.billing.catalog.api.boilerplate.PlanImp;
+
+@JsonDeserialize( builder = PlanModel.Builder.class )
+public class PlanModel extends PlanImp {
```

Comment:
Generic question for all files under `org.killbill.billing.plugin.catalog.test.models.api`: it seems like these models all extends the api POJO generated. 
* Why do we need these extras POJO, can't we just use the generated ones (e.g  `PlanImp`)
* Did you write all those files by hand or did you also use a tool (`api-pojos`) to generate those?

Reply:
A lot of the catalog entities have circular references to each other and to the catalog itself so I had to re-implement the 
`equal(java.lang.Object)`, `hashCode()` and `toString()` to avoid going into infinite recursion.

Reply:
😞